coderd/workspaceapps/provider.go

```diff
@@ -11,10 +13,45 @@ import (
 	"github.com/coder/coder/codersdk"
 )
 
-// Provider provides authentication and authorization for workspace apps.
-// TODO(@deansheather): also provide workspace apps as a whole to remove all app
-// code from coderd.
-type Provider struct {
+/*
+POST /api/v2/moons/app-auth-ticket
+
+{
+	"session_token": "xxxx",
+	"request": { ... }
+}
+
+type moonRes struct {
+	Ticket *Ticket
+	TicketStr string
+}
+*/
+
+// TicketProvider provides workspace app tickets.
+//
+// write a funny comment that says a ridiculous amount of fees will be incurred:
+//
+// Please keep in mind that all transactions incur a service fee, handling fee,
+// order processing fee, delivery fee,
```

Comment:
I might rename this from `TicketProvider` to something like `RefreshingTokenProvider` or something. Just because we already use the token wording.

Reply:
We call the session token a token and we already call the ticket a ticket. I'd rather use a different term than token to avoid people thinking it's the same thing as a session token.

Reply:
This is also invisible to the end user so the naming of it doesn't matter that much. I like TicketProvider more than RefreshingTokenProvider

Reply:
Well, this is really a fast-expiring token, so I'm just trying to reduce the number of terms people need to consume to understand what's going on.

I'm proposing renaming Ticket to something like `SignedToken` (or something else token related, so that it doesn't confuse people).

Reply:
`SignedToken` is a valid name imo.

Reply:
Maybe we do `SignedExpiringToken` to make it verbose but super obvious!

Reply:
SignedExpiringToken isn't obvious because it doesn't say what it's for. SignedAppToken would be better

Reply:
Renamed to `SignedToken` and the cookie says `coder_devurl_signed_app_token`.